crates/uv/src/commands/tool/uninstall.rs

```diff
@@ -37,6 +37,28 @@ pub(crate) async fn uninstall(
         Err(err) => return Err(err.into()),
     };
 
+    // Perform the uninstallation.
+    do_uninstall(&installed_tools, name, printer).await?;
+
+    // Clean up any empty directories.
+    if uv_fs::directories(installed_tools.root()).all(|path| uv_fs::is_temporary(&path)) {
+        fs_err::tokio::remove_dir_all(&installed_tools.root()).await?;
+        if let Some(top_level) = installed_tools.root().parent() {
+            if uv_fs::directories(top_level).all(|path| uv_fs::is_temporary(&path)) {
+                fs_err::tokio::remove_dir_all(top_level).await?;
+            }
+        }
+    }
```

Comment:
We clean up the directories, if empty, such that if you `uninstall --all`, we remove `/Users/crmarsh/Library/Application Support/uv/tools`, and then `/Users/crmarsh/Library/Application Support/uv` if _that_ directory is empty.